### ISSUE TITLE
NVDA & web application role cannot navigate PDOM when tab focused on solute adjuster or protein in membrane

### ISSUE THREAD

**Issue by brettfiedler** [phetsims/membrane-transport#376]
Experienced on NVDA (latest), Win11 (latest), Chrome (latest)

With NVDA, if you have tab navigated to one of the solute adjusters or a protein on the membrane, you cannot access the PDOM fully using browse mode. You see the accessible name repeated when navigating up and down, but no help text. You can tab away from it, but you cannot get away from it with the virtual cursor (A Virtual Cursor Trap?).

@terracoda says this is likely a problem with how NVDA interacts with the web-application role. 

<img width="636" height="891" alt="Image" src="https://github.com/user-attachments/assets/a7ec8ee3-9b5a-4947-85a1-dbfb6e5c0666" />

**Comment by jessegreenberg** [comment]
I am not sure I am seeing the same thing, but I just wanted to note that "browse" and "forms" mode are the two distinct modes and should behave differently.

In forms mode, you should be able to press the arrow keys + AD keys to add/remove solute. This is working for me in Chrome + Firefox. You shouldn't need to press NVDA key at all (and I wouldn't expect that to work).

In browse mode, arrow keys should move the virtual cursor. Indeed, it seems to get stuck on the accessibleName of the solute adjuster. I am not sure why and will take a look.

**Comment by jessegreenberg** [comment]
I can see this happen in a smple HTML snippet:
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Simple range input</title>
</head>
<body>
<div tabindex="0" aria-roledescription="solute adjuster" role="application" aria-label="Oxygen Molecules Outside Cell">Oxygen Molecules Outside Cell
</div>

<p>Here is another paragraph</p>
<h2>Here is a heading</h2>
</body>
</html>
```

**Comment by jessegreenberg** [comment]
I don't see a fix yet. In this test, I add content under the `application` element. NVDA can easily read it but still gets stuck inside the `application` element.

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Simple range input</title>
</head>
<body>
<p>Something before the div</p>
<div tabindex="0" aria-roledescription="solute adjuster" role="application" aria-label="Oxygen Molecules Outside Cell">
  <span>Oxygen Molecules Outside Cell</span>
  <p>Something inside the div</p>
</div>

<p>Here is another paragraph</p>
<h2>Here is a heading</h2>
</body>
</html>
```

o3 states that this is expected behavior.

>As soon as it sees a focusable element whose accessibility role is “application”, it treats that subtree as a separate document, the same way it does for an <iframe> or a PDF object.
> When you press NVDA+Space to force Browse mode, NVDA builds a small virtual-buffer that contains only the nodes that belong to that application subtree.
> Arrow-key navigation is always confined to the current virtual-buffer, so you cannot move past the boundary with the normal cursor keys.

> How to get out

> - Use NVDA+Ctrl+Up-Arrow (move to previous document) or NVDA+Ctrl+Down-Arrow (next document).
> - Alternatively, Tab / Shift+Tab to shift real keyboard focus outside the widget and then toggle Browse mode again if you want.
> - Object-navigation commands (NVDA+NumPad 8 / 2 / 4 / 6) will also let you move to the parent object and then back into the main document.

But only item 2 works.

**Comment by brettfiedler** [comment]
We should probably report an issue to their repo: https://github.com/nvaccess/nvda/issues

Seems active enough!

**Comment by brettfiedler** [comment]
I was just playing around with this on ChromeOS Flex and their ChromeVox screen reader and did not have a problem here, just as a data point. In that case, you need to hold the "special key" and move around the PDOM, but when releasing it on the oslute adjusters, I could then add and remove, then go about my way back through the PDOM.

**Comment by jessegreenberg** [comment]
OK, I created https://github.com/nvaccess/nvda/issues/18644. I also added this to https://github.com/phetsims/qa/blob/main/documentation/accessibility-bugs.md. Closing.
